src/main/java/fi/aalto/cs/apluscourses/model/Exercise.java

```diff
@@ -29,12 +29,22 @@ public static Exercise fromJsonObject(@NotNull JSONObject jsonObject) {
     return new Exercise(id, name);
   }
 
+  @NotNull
+  public String getName() {
+    return name;
+  }
+
   public long getId() {
     return id;
   }
 
-  @NotNull
-  public String getName() {
-    return name;
+  @Override
+  public int hashCode() {
+    return Long.hashCode(id);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    return obj instanceof Exercise && ((Exercise) obj).getId() == getId();
```

Comment:
the latter one is `this.getId()`, right?

Reply:
Yes.